### PR TITLE
Billing & currency jobs - Send billing and currency data to Sap

### PR DESCRIPTION
**Pull request by nreal**
# Background
When Billing job is executed,  the job consumes the Doppler database about the user billing information and then send this data to the SAP system
Billing example:
`[
  {
    "Id": 63,
    "PlanType": 0,
    "CreditsOrSubscribersQuantity": 0,
    "IsCustomPlan": false,
    "IsPlanUpgrade": false,
    "Currency": 0,
    "PeriodMonth": 4,
    "PeriodYear": 2020,
    "PlanFee": 2.0,
    "ExtraEmailsPeriodMonth": 3,
    "ExtraEmailsPeriodYear": 2020,
    "ExtraEmailsFee": 0.0
  },
  {
    "Id": 69191,
    "PlanType": 2,
    "CreditsOrSubscribersQuantity": 0,
    "IsCustomPlan": false,
    "IsPlanUpgrade": false,
    "Currency": 0,
    "Periodicity": 0,
    "PeriodMonth": 4,
    "PeriodYear": 2020,
    "PlanFee": 510.0,
    "ExtraEmailsFeePerUnit": 0.0026,
    "ExtraEmailsPeriodMonth": 3,
    "ExtraEmailsPeriodYear": 2020,
    "ExtraEmailsFee": 0.0
  }]`

Currency example:
`[
  {
    "Entity": {
      "Date": "2020-04-23",
      "SaleValue": 67.7500,
      "BuyValue": 62.7500,
      "CurrencyName": "Peso Argentino",
      "CurrencyCode": "ARS"
    }
  }
]`

Can you review it?

### CODE REVIEW

File: DopplerJobTest/DopplerSapServiceTests.cs [hunk 1]
```diff
@@ -28,7 +28,8 @@ public DopplerSapServiceTests()
             _dopplerSapServiceSettingsMock.Setup(x => x.CurrentValue)
                 .Returns(new DopplerSapConfiguration
                 {
-                    Url = "https://localhost:5001/SetCurrencyRate"
+                    CurrencyEndpoint = "https://localhost:5001/SetCurrencyRate",
+                    BillingEndpoint = "https://localhost:5001/Billing/CreateBillingRequest"
```
Comment (fgchaio) [inline DopplerJobTest/DopplerSapServiceTests.cs:32]: Minor change. In the local environment we are using the 65100 port.
```suggestion
                    CurrencyEndpoint = "https://localhost:65100/SetCurrencyRate",
                    BillingEndpoint = "https://localhost:65100/Billing/CreateBillingRequest"
```

File: DopplerJobTest/DopplerSapServiceTests.cs [hunk 2]
```diff
@@ -28,8 +28,8 @@ public DopplerSapServiceTests()
             _dopplerSapServiceSettingsMock.Setup(x => x.CurrentValue)
                 .Returns(new DopplerSapConfiguration
                 {
-                    CurrencyEndpoint = "https://localhost:5001/SetCurrencyRate",
-                    BillingEndpoint = "https://localhost:5001/Billing/CreateBillingRequest"
+                    CurrencyEndpoint = "https://localhost:65100/SetCurrencyRate",
```
Comment (swidzinskiMS) [inline DopplerJobTest/DopplerSapServiceTests.cs:31]:
```suggestion
                    CurrencyEndpoint = "https://localhost:65100/Billing/SetCurrencyRate"",
```

Comment (swidzinskiMS) [reply DopplerJobTest/DopplerSapServiceTests.cs:31]: That's the endpoint now

**Comment by swidzinskiMS** [comment]
I see the object you are sending for the currency endpoint is not the one I've sent you. It should be 
{
    "Currency": "MXN",
    "Rate": "14",
    "RateDate": "2020-02-17"
}
In ARS currency send saleValue

**Comment by nreal** [comment]
@swidzinskiMS can you check it pr https://github.com/FromDoppler/doppler-currency/pull/20 ?

**Pull request merged**